app/src/androidTest/java/com/example/sharingang/FakeRepositoryModule.kt

```diff
@@ -71,13 +71,16 @@ object FakeRepositoryModule {
     @Provides
     fun provideFirebaseAuth(): FirebaseAuth {
         val auth = FirebaseAuth.getInstance()
-        if (BuildConfig.DEBUG) {
-            Log.d("RepositoryModule", "Using FirebaseAuth emulator.")
-            auth.useEmulator("10.0.2.2", 9099)
-        } else {
-            Log.d("RepositoryModule", "Using production FirebaseAuth.")
-        }
+        auth.useEmulator("10.0.2.2", 9099)
         return auth
     }
+
+    @Singleton
+    @Provides
+    fun provideFirebaseAuthUI(): AuthUI {
+        val authUI = AuthUI.getInstance()
+        authUI.useEmulator("10.0.2.2", 9099)
+        return authUI
+    }
 }
```

Comment:
Pretty nice! 🥇

Reply:
And you'll see when logging in during development, you no longer need a real Google Account. You can generate fake ones easily.
<img src="https://user-images.githubusercontent.com/1939395/117376930-6cffca00-aed2-11eb-8456-cc4a4559c9bc.png" width="300">

Reply:
Awesome! This will truly prove very useful, well done! ^^